### DISCUSSION TITLE
Can't use structuredClone from a store state property

### DISCUSSION THREAD

**Discussion by sergiomartindev** [vuejs/pinia#1412] [Help and Questions]
I'm trying to create a deep clone (via the global structuredClone method) of an array stored in a store' state; but it throws the following error:

```
Uncaught (in promise) DOMException: Failed to execute 'structuredClone' on 'Window': [object Array] could not be cloned.
```

I'm aware that it might be caused by the own behavior of the structuredClone method, which doesn't support proxies. So my question is, is there any alternative way to tackle this problem?

I've used the toRaw method, which seems to be valid, so the result looks this way:

```
import { toRaw } from 'vue';

const myClone = structuredClone(toRaw(myStoredArray));
```

Thank you.

**Comment by posva** [comment]
The error looks like you have a property in your state that contains the global window object. What did you store in your state?
